atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SerializableTransactionManager.java

```diff
@@ -37,25 +38,30 @@
 public class SerializableTransactionManager extends SnapshotTransactionManager {
 
     public static class InitializeCheckingWrapper extends AutoDelegate_SerializableTransactionManager {
-        private SerializableTransactionManager manager;
+        private final SerializableTransactionManager manager;
+        private final AsyncInitializer prerequisite;
+
         private volatile boolean isInitialized = false;
 
-        public InitializeCheckingWrapper(SerializableTransactionManager manager) {
+        public InitializeCheckingWrapper(SerializableTransactionManager manager,
+                AsyncInitializer prerequisite) {
             this.manager = manager;
+            this.prerequisite = prerequisite;
         }
 
         @Override
         public SerializableTransactionManager delegate() {
             if (!isInitialized) {
-                try {
-                    // SerializableTransactionManager should throw until the underlying KVS is initialized.
-                    // TODO(ssouza): replace with KVS healthcheck status when that gets implemented.
-                    manager.getKeyValueService().getClusterAvailabilityStatus();
-                } catch (NotInitializedException e) {
-                    log.info("The KeyValueService is not initialized yet!");
-                    throw e;
+                // Verify that the various dependent services have been initialized
+                if (manager.getKeyValueService().isInitialized()
```

Comment:
are you planning to add an `isInitialized()` method on `TransactionManager`?

Reply:
done